scapy/utils.py

```diff
@@ -429,17 +429,22 @@ class ContextManagerSubprocess(object):
     >>>     subprocess.Popen(["unknown_command"])
 
     """
-    def __init__(self, name):
+    def __init__(self, name, prog, _raise=True):
         self.name = name
+        self.prog = prog
+        self._raise = _raise
 
     def __enter__(self):
         pass
 
     def __exit__(self, exc_type, exc_value, traceback):
-        if exc_type == OSError:
-            msg = "%s: executing %r failed"
-            log_runtime.error(msg, self.name, conf.prog.wireshark, exc_info=1)
-            return True  # Suppress the exception
+        if exc_type in [OSError, TypeError]:
```

Comment:
It would be better to check `issubclass()`, or `isinstance()` with `exc_value`, since Python 3 has specific subclasses for `OSError` (`FileNotFoundError`, for example), and I guess in that case this test would fail.

Reply:
You're right. Updated